src/Nethermind/Nethermind.Serialization.Ssz/Ssz.Encode.cs

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections;
+
+namespace Nethermind.Serialization.Ssz;
+
+/// <summary>
+/// https://github.com/ethereum/eth2.0-specs/blob/dev/specs/simple-serialize.md#simpleserialize-ssz
+/// </summary>
+public static partial class Ssz
+{
+    public static void Encode(Span<byte> span, BitArray? vector)
+    {
+        if (vector is null)
+        {
+            return;
+        }
+        int byteLength = (vector.Length + 7) / 8;
+        byte[] bytes = new byte[byteLength];
+        vector.CopyTo(bytes, 0);
```

Comment:
Do we really have to copy? We could aquire internal array as ReadOnlySpan through reflection, check how we cache delegates in `ConcurrentDictionaryLock`